workbench/tests/renderer/plugin.test.js

```diff
@@ -45,7 +47,95 @@ describe('Add plugin modal', () => {
     getInvestModelIDs.mockResolvedValue({});
   });
 
-  test('Interface to add a plugin', async () => {
+  describe('"Add a plugin" form validation', () => {
+    let spy;
+
+    beforeEach(async () => {
+      spy = ipcRenderer.invoke.mockImplementation((channel, setting) => {
+        if (channel === ipcMainChannels.GET_SETTING) {
+          if (setting === 'plugins') {
+            return Promise.resolve({
+              foo: {
+                modelTitle: 'Foo',
+                type: 'plugin',
+              },
+            });
+          }
+        } else if (channel === ipcMainChannels.HAS_MSVC) {
+          return Promise.resolve(true);
+        }
+        return Promise.resolve();
+      });
+    });
+
+    test('Should render an error on submit if git URL is empty', async () => {
+      const {
+        findByText, findByLabelText, findByRole,
+      } = render(<App />);
+
+      await userEvent.click(await findByRole('button', { name: 'menu' }));
+      const managePluginsButton = await findByText(/Manage plugins/i);
+      userEvent.click(managePluginsButton);
```

Comment:
I think `userEvent` actions always returns a promise, so best to `await` it in all these cases.

Reply:
I have also been around and around on this in the past and don't fully understand it. The version 14 release notes state that the APIs always return a promise, so `awaiting` seems appropriate. https://github.com/testing-library/user-event/releases/tag/v14.0.0

But it's possible that some tests that did not `await` would proceed to `await` something else, like a `findBy` query, that allowed the promise to resolve in time. Not awaiting the userEvent actions could sometimes lead to the dreaded `not wrapped in act()` warning from React. Adding `await` has resolved that for me in the past.

I also wonder if the resolution of the promise also depends on the event handler, which maybe makes the 'Add' click a unique case.